app/src/main/java/org/simple/clinic/drugs/selection/entry/CustomPrescriptionEntryUiRenderer.kt

```diff
@@ -8,9 +8,28 @@ class CustomPrescriptionEntryUiRenderer(val ui: CustomPrescriptionEntryUi) : Vie
   private val drugNameValueCallback = ValueChangedCallback<Boolean>()
 
   override fun render(model: CustomPrescriptionEntryModel) {
+    showDefaultDosagePlaceholder(model.dosage, model.dosageHasFocus)
+
     if (model.drugName == null) return
 
     val isDrugNameNotBlank = model.drugName.isNotBlank()
     drugNameValueCallback.pass(isDrugNameNotBlank, ui::setSaveButtonEnabled)
+
+  }
+
+  /**
+   * The dosage field shows a default text as "mg". When it is focused, the cursor will
+   * by default be moved to the end. This will force the user to either move the cursor
+   * to the end manually or delete everything and essentially making the placeholder
+   * useless. As a workaround, we move the cursor to the starting again.
+   */
+  private fun showDefaultDosagePlaceholder(dosage: String?, dosageHasFocus: Boolean?) {
+    if (dosage == null || dosageHasFocus == null) return
+
+    when {
+      dosage.isBlank() && dosageHasFocus -> ui.setDrugDosageText(DOSAGE_PLACEHOLDER)
+      dosage.trim() == DOSAGE_PLACEHOLDER && dosageHasFocus.not() -> ui.setDrugDosageText("")
+      dosage.trim() == DOSAGE_PLACEHOLDER && dosageHasFocus -> ui.moveDrugDosageCursorToBeginning()
+    }
```

Comment:
Yes. You are right about this. I was planning on doing this refactor in the end but forgot about it. Should have added a `TODO`. 
I will do this now.

Reply:
This feature has broken as part of the refactor, so if moving this out is simpler and fixes the problem, we can do it now.

Otherwise, I am fine with fixing the problem as is and raising a ticket to do this later since it's already a large-ish PR.

Reply:
I will have to make this change in order to make it work. But it will be difficult to change this commit now.
I will see if I can drop this commit and create a new commit and reorder it.

Reply:
I ended up fixing the commit.